src/Indicator.vala

```diff
@@ -89,7 +133,10 @@ public class DateTime.Indicator : Wingpanel.Indicator {
             GLib.Source.remove (update_events_idle_source);
         }
 
-        update_events_idle_source = GLib.Idle.add (update_events);
+        update_events_idle_source = GLib.Idle.add (() => {
+            update_events ();
+            return false;
```

Comment:
It looks like `update_events` is a boolean method, so why do we return `false` here? It seems like the previous syntax already made sense